File: perl/variable/boolean_value/src/variable/variable.pl
#!/usr/bin/perl

# 0は偽.
$var1 = 0;
if ($var1){
  print "true\n";
}
else{
  print "false\n";
}

# 空文字列""は偽.
$var2 = "";
if ($var2){
  print "true\n";
}
else{
  print "false\n";
}

# 空リスト()は偽.
@ary = (); # 配列@aryに空リスト.
if (@ary){
  print "true\n";
}
else{
  print "false\n";
}
$var3 = (); # 変数$var3に空リスト.
if ($var3){
  print "true\n";
}
else{
  print "false\n";
}

# 未定義値は空リスト.
$var4 = undef; # $var4にundef.
if ($var4){
  print "true\n";
}
else{
  print "false\n";
}

File: ruby/variable/local_variable/src/variable/variable.rb
#!/usr/bin/ruby

# ローカル変数var1の値を出力するoutputVar1.
def outputVar1
  var1 = 10; # var1を10で初期化.
  puts(var1); # putsでvar1を出力. 
end

# outputVar1を呼ぶ.
outputVar1
puts(var1) # ここでvar1を使おうとしてもスコープ外.
